src/app/api/game-sessions/[gameSessionDate]/route.ts

```diff
@@ -0,0 +1,172 @@
+import { and, eq, gte, lte, or } from "drizzle-orm";
+import { z } from "zod";
+
+import { db } from "@/lib/db";
+import {
+  gameSessionExceptions,
+  gameSessions,
+  gameSessionSchedules,
+  semesters,
+  weekdayEnum,
+} from "@/lib/db/schema";
+import {
+  insertGameSessionExceptionSchema,
+  insertGameSessionSchema,
+  updateGameSessionSchema,
+} from "@/lib/validators";
+
+const routeContextSchema = z.object({
+  params: z.object({
+    gameSessionDate: z.coerce.date(),
+  }),
+});
+
+export async function GET(
+  _req: NextRequest,
+  context: z.infer<typeof routeContextSchema>,
+) {
+  try {
+    const result = routeContextSchema.safeParse(context);
+    if (!result.success)
+      return new Response("Invalid date provided in the request", {
+        status: 400,
+      });
+
+    const {
+      params: { gameSessionDate },
+    } = result.data;
+
+    const gameSession = await db.query.gameSessions.findFirst({
+      where: eq(gameSessions.date, gameSessionDate),
+    });
+
+    if (gameSession) {
+      return NextResponse.json(gameSession, { status: 200 });
+    }
+
+    const [gameSessionSchedule] = await db
+      .select()
+      .from(gameSessionSchedules)
+      .innerJoin(
+        semesters,
+        or(
+          and(
+            gte(semesters.startDate, gameSessionDate),
+            lte(semesters.breakStart, gameSessionDate),
+          ),
+          and(
+            gte(semesters.breakEnd, gameSessionDate),
+            lte(semesters.endDate, gameSessionDate),
+          ),
+        ),
+      )
+      .where(
+        eq(
+          gameSessionSchedules.weekday,
+          weekdayEnum.enumValues[gameSessionDate.getDay()],
+        ),
+      );
+
+    if (!gameSessionSchedule) {
+      return new Response(null, { status: 204 });
+    }
+
+    const gameSessionException = await db.query.gameSessionExceptions.findFirst(
+      {
+        where: eq(gameSessionExceptions.gameSessionDate, gameSessionDate),
+      },
+    );
+
+    if (gameSessionSchedule && !gameSessionException) {
+      return NextResponse.json(gameSessionSchedule, { status: 200 });
+    }
+  } catch {
+    return new Response("Internal Server Error", { status: 500 });
+  }
+}
+
+export async function DELETE(
+  req: NextRequest,
+  context: z.infer<typeof routeContextSchema>,
+) {
+  try {
+    const result = routeContextSchema.safeParse(context);
+    if (!result.success)
+      return new Response("Invalid date provided in the request", {
+        status: 400,
+      });
+
+    const {
+      params: { gameSessionDate },
+    } = result.data;
+
+    const gameSession = await db.query.gameSessions.findFirst({
+      where: eq(gameSessions.date, gameSessionDate),
+    });
+
+    if (gameSession?.id) {
+      // Delete the gameSession and return
+      await db
+        .delete(gameSessions)
+        .where(eq(gameSessions.id, gameSession.id))
+        .returning();
+
+      return new Response(null, { status: 204 });
+    }
```

Comment:
i think a better way to go about this is to delete the gameSession by date instead of id with returning.

If the return is null, then we can create a exception, otherwise 204.